pkg/network/http/shared_libraries.go

```diff
@@ -167,7 +179,14 @@ func (w *soWatcher) sync() {
 			if r.re.MatchString(path) {
 				// new library detected
 				// register it
-				w.registry.register(path, r)
+				for _, pidPath := range lib.PidsPath {
+					if pidPathArray := strings.Split(pidPath, string(os.PathSeparator)); len(pidPathArray) > 0 {
+						pidStr := pidPathArray[len(pidPathArray)-1]
+						if pid, err := strconv.Atoi(pidStr); err != nil {
```

Comment:
log on error, as it should not happen and if it does -> there is a problem with the reported `PidsPath`